lib/elixir/lib/calendar/time.ex

```diff
@@ -469,19 +476,22 @@ defmodule Time do
       ~T[17:30:00.000000]
       iex> Time.add(~T[11:00:00.005], 2400)
       ~T[11:40:00.005000]
-      iex> Time.add(~T[00:00:00], 86_399_999, :millisecond)
-      ~T[23:59:59.999000]
       iex> Time.add(~T[17:10:05], 86400)
       ~T[17:10:05.000000]
       iex> Time.add(~T[23:00:00], -60)
       ~T[22:59:00.000000]
+      iex> Time.add(~T[00:00:00], 86_399_999, :millisecond)
+      ~T[23:59:59.999000]
+      iex> Time.add(~T[10:00:00], 1, :minute)
+      ~T[10:01:00.000000]
+      iex> Time.add(~T[10:00:00], 1, :hour)
+      ~T[11:00:00.000000]
 
   """
   @doc since: "1.6.0"
-  @spec add(Calendar.time(), integer, System.time_unit()) :: t
+  @spec add(Calendar.time(), integer, time_unit()) :: t()
   def add(%{calendar: calendar} = time, number, unit \\ :second) when is_integer(number) do
-    number = System.convert_time_unit(number, unit, :microsecond)
-    total = time_to_microseconds(time) + number
+    total = time_to_microseconds(time) + convert_time_unit(number, unit)
```

Comment:
I would be more specific about the final unit:

```suggestion
    total = time_to_microseconds(time) + time_unit_to_microseconds(number, unit)
```

or:

```suggestion
    total = time_to_microseconds(time) + convert_time_unit(number, unit, :microsecond)
```


Thoughts? 🙂

Reply:
I like the consistency of `time_unit_to_microseconds/2` 👍